### PR TITLE
Convert the build_release.sh script to a dart script

### PR DESCRIPTION
**Pull request by kenzieschmoll**
work towards: https://github.com/flutter/devtools/issues/6324

@DanTup this should enable this script to work on windows.

### CODE REVIEW

File: tool/lib/commands/build_release.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      ],
+      workingDirectory: repo.devtoolsAppDirectoryPath,
+    );
+
+    final canvaskitDir = Directory(
+      path.join(repo.devtoolsAppDirectoryPath, 'build', 'web', 'canvaskit'),
+    );
+    for (final file in canvaskitDir.listSync()) {
+      if (RegExp(r'canvaskit\..*').hasMatch(file.path)) {
+        await processManager.runProcess(CliCommand('chmod 0755 ${file.path}'));
```
Comment (kenzieschmoll) [inline tool/lib/commands/build_release.dart:80]: @DanTup do you know the preferred way to do this on windows? I searched on Stack overflow but didn't see one definitive answer

Comment (DanTup) [reply tool/lib/commands/build_release.dart:80]: Is it necessary?

I can test it tomorrow when back on PC, but for the chmod issue in the build_and_copy script I just removed it and didn't see any issues. It's not obvious to me what this is doing though - why is the permissions change necessary for non-Windows?

Comment (kenzieschmoll) [reply tool/lib/commands/build_release.dart]: I added a TODO to investigate whether changing file permissions is necessary on windows or at all anymore.

File: tool/lib/commands/build_release.dart [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    final useLocalFlutter = argResults![_useLocalFlutterFlag];
+    final updatePerfetto = argResults![_updatePerfettoFlag];
+
+    if (!useLocalFlutter) {
+      logStatus('updating tool/flutter-sdk to the latest flutter candidate');
+      await processManager.runProcess(CliCommand.tool('update-flutter-sdk'));
+    }
+
+    if (updatePerfetto) {
+      logStatus('updating the bundled Perfetto assets');
```
Comment (elliette) [inline tool/lib/commands/build_release.dart:56]: suggestion: should the comment for now to be "skipping updating the bundled Perfetto assets" so we aren't confused why the perfetto assets don't get updated?

**Review by elliette — APPROVED** [review]

**Pull request merged**